plugins/chat/assets/javascripts/discourse/components/chat/thread-list.js

```diff
@@ -109,17 +110,27 @@ export default class ChatThreadList extends Component {
     restoredOriginalMessageThread.originalMessage.deletedAt = null;
   }
 
-  @action
-  loadThreads() {
-    this.loading = true;
-    this.args.channel.threadsManager.index(this.args.channel.id).finally(() => {
-      this.loading = false;
+  @cached
+  get threadsCollection() {
+    return this.chatApi.threads(this.args.channel.id, this.handleLoadedThreads);
+  }
+
+  @bind
+  handleLoadedThreads(result) {
+    return result.threads.map((thread) => {
+      return this.threadsManager.add(this.args.channel, thread, {
+        replace: true,
+      });
     });
+
+    this.chatTrackingStateManager.setupChannelThreadState(
```

Comment:
I wrote up the bug report below, but then while investigating it I found the issue and pushed a fix. **Just leaving it here for posterity, no need to do a further fix!!**. It was just because the `return result.threads.map` above was returning early and also `chatTrackingStateManager` was not added as a service here.

------

I think this maybe is messing up the tracking state. To reproduce:

1. Send a message as another user in a thread you are tracking
2. See that there is the unread thread indicator in the channel header
3. Open the thread list. You will see the unread indicator on the list item briefly then it disappears

![image](https://github.com/discourse/discourse/assets/920448/a1bf3859-10f2-4701-8266-6b512349edf0)

![image](https://github.com/discourse/discourse/assets/920448/faedddaf-efd3-4f0a-bf72-9e2c3df6caf8)

If you open the thread in the list the unread indicator in the channel header will disappear because the thread is read.